spyder/plugins/profiler/plugin.py

```diff
@@ -4,113 +4,138 @@
+from spyder.api.translations import get_translation
 from spyder.plugins.profiler.confpage import ProfilerConfigPage
-from spyder.plugins.profiler.widgets.profilergui import (ProfilerWidget,
+from spyder.plugins.profiler.widgets.main_widget import (ProfilerWidget,
+                                                         ProfilerWidgetActions,
                                                          is_profiler_installed)
+from spyder.preferences.runconfig import get_run_configuration
 
+# Localization
+_ = get_translation('spyder')
 
-if is_dark_interface():
-    MAIN_TEXT_COLOR = 'white'
-else:
-    MAIN_TEXT_COLOR = '#444444'
 
+# --- Constants
+# ----------------------------------------------------------------------------
+class ProfilerActions:
+    ProfileCurrentFile = 'profile_current_filename_action'
 
-class Profiler(SpyderPluginWidget):
-    """Profiler (after python's profile and pstats)."""
 
-    CONF_SECTION = 'profiler'
-    CONFIGWIDGET_CLASS = ProfilerConfigPage
-    CONF_FILE = False
+# --- Plugin
+# ----------------------------------------------------------------------------
+class Profiler(SpyderDockablePlugin):
+    """
+    Profiler (after python's profile and pstats).
+    """
 
-    def __init__(self, parent=None):
-        SpyderPluginWidget.__init__(self, parent)
+    NAME = 'profiler'
+    REQUIRES = [Plugins.Editor, Plugins.VariableExplorer]
+    TABIFY = Plugins.Help
+    WIDGET_CLASS = ProfilerWidget
+    CONF_SECTION = NAME
+    CONF_WIDGET_CLASS = ProfilerConfigPage
+    CONF_FILE = False
 
-        max_entries = self.get_option('max_entries', 50)
-        self.profiler = ProfilerWidget(self, max_entries,
-                                       options_button=self.options_button,
-                                       text_color=MAIN_TEXT_COLOR)
+    # --- Signals
+    # ------------------------------------------------------------------------
+    sig_started = Signal()
+    """This signal is emitted to inform the profiling process has started."""
 
-        layout = QVBoxLayout()
-        layout.addWidget(self.profiler)
-        self.setLayout(layout)
+    sig_finished = Signal()
+    """This signal is emitted to inform the profile profiling has finished."""
 
-    #------ SpyderPluginWidget API ---------------------------------------------    
-    def get_plugin_title(self):
-        """Return widget title"""
+    # --- SpyderDockablePlugin API
+    # ------------------------------------------------------------------------
+    def get_name(self):
         return _("Profiler")
 
-    def get_plugin_icon(self):
-        """Return widget icon"""
-        path = osp.join(self.PLUGIN_PATH, self.IMG_PATH)
-        return ima.icon('profiler', icon_path=path)
+    def get_description(self):
+        return _("Profile your scripts and find bottlenecks.")
+
+    def get_icon(self):
+        path = os.path.join(self.get_path(), self.IMG_PATH)
```

Comment:
Please revert this and use `osp` instead.

Reply:
Fixed